taui/src/components/edit-profile.js

```diff
@@ -482,7 +482,7 @@ class EditProfile extends PureComponent<Props> {
     const options = PROFILE_DESTINATION_TYPES.map((key) => {
       // expects each type in constants to have a label in messages
       const messageKey = 'TripPurpose.' + key
-      return <option key={key}>{t(messageKey)}</option>
+      return <option key={messageKey} value={messageKey}>{t(messageKey)}</option>
```

Comment:
This works but it has the side-effect (as you noticed) that existing profiles become incompatible with the new code and need to be migrated. You should be able to achieve the same change, with fewer side-effects, by doing this instead:
```suggestion
      return <option key={key} value={key}>{t(messageKey)}</option>
```